### PR TITLE
fixed to be able to handle NoneType

### PR DESCRIPTION
**Pull request by tazhfe**

### CODE REVIEW

**Comment by tazhfe** [comment]
Video demo:
https://drive.google.com/file/d/1Gpv-QSt99X8PuPIuyXYKEDX3uya0Co1U/view?usp=drive_link

File: scripts/temp_pub.py [hunk 1]
```diff
@@ -12,17 +12,28 @@ def internal_temperature_publisher():
     msg_temp = Temperature()
     msg_temp.header.frame_id = "internal_temperature_sensor_link"
     rate = rospy.Rate(10)
+    prev_temp = 0
```
Comment (arwinder-jaspal) [inline scripts/temp_pub.py]: is this a float or an int?

File: scripts/temp_pub.py [hunk 2]
```diff
@@ -12,17 +12,28 @@ def internal_temperature_publisher():
...
+
     while not rospy.is_shutdown():
         try:
             msg_temp.header.stamp = rospy.Time.now()
             sensor_data = temper.get_temperature()
-            msg_temp.temperature = float(sensor_data[0])
+
+            if sensor_data is not None and sensor_data[0] is not None:
+                msg_temp.temperature = float(sensor_data[0])
```
Comment (arwinder-jaspal) [inline scripts/temp_pub.py:23]: is the dynamic casting required? What is the data type of the incoming data?

File: scripts/temp_pub.py [hunk 3]
```diff
@@ -12,17 +12,28 @@ def internal_temperature_publisher():
...
+                prev_temp = float(sensor_data[0])
+            else:
+                msg_temp.temperature = float(prev_temp)
+                rospy.loginfo("Error reading temperature, displaying previously read temperature")
+            
+            print(msg_temp)
```
Comment (arwinder-jaspal) [inline scripts/temp_pub.py]: remove print state. I would recommend using rospy.logdebug for debugging purpose

File: scripts/temp_pub.py [hunk 4]
```diff
@@ -12,17 +12,28 @@ def internal_temperature_publisher():
...
             msg_temp.variance = 0
             pub.publish(msg_temp)
             rate.sleep()
         except SystemExit:
             print("Process has exited unexpectedly")
```
Comment (arwinder-jaspal) [inline scripts/temp_pub.py:34]: Can we update this to a log error or warn?

**Review by arwinder-jaspal — CHANGES_REQUESTED** [review]
Overall, good PR. I made some minor comments. Let me know if you need any help

File: scripts/temp_pub.py (hunk 2 above)
Comment (tazhfe) [reply scripts/temp_pub.py:23]: yes, the incoming data type for temperature is str

File: scripts/temp_pub.py (hunk 1 above)
Comment (tazhfe) [reply scripts/temp_pub.py]: float, will assign 0.0 for clarity

File: scripts/temp_pub.py (hunk 4 above)
Comment (tazhfe) [reply scripts/temp_pub.py:34]: Will update to logerr

File: scripts/temp_pub.py (hunk 2 above)
Comment (arwinder-jaspal) [reply scripts/temp_pub.py:23]: the dynamic casting should happen only once then

**Review by arwinder-jaspal — CHANGES_REQUESTED** [review]
added one more comment. should be good to go after that

File: scripts/temp_pub.py (hunk 2 above)
Comment (arwinder-jaspal) [reply scripts/temp_pub.py:23]: is this fixed?

Comment (tazhfe) [reply scripts/temp_pub.py:23]: yes, didn't sync changes previously. changes are synced now

**Comment by arwinder-jaspal** [comment]
can you share the latest testing video here?

**Comment by tazhfe** [comment]
latest testing video here: 
https://drive.google.com/file/d/1LuxyqyhYr6h15LJxboBOu33dxPTn-Zg-/view?usp=drive_link

**Review by arwinder-jaspal — APPROVED** [review]

**Pull request merged**